includes/embeds/class-amp-core-block-handler.php

```diff
@@ -211,6 +212,41 @@ public function ampify_video_block( $block_content, $block ) {
 		return $block_content;
 	}
 
+	/**
+	 * Ampify file block.
+	 *
+	 * Inject an inline style attribute to handle PDF embeds not being displayed in some themes (eg. Twenty Twenty).
+	 *
+	 * @see \AMP_Object_Sanitizer::sanitize_pdf()
+	 *
+	 * @param string $block_content The block content about to be appended.
+	 * @param array  $block         The full block, including name and attributes.
+	 * @return string Filtered block content.
+	 */
+	public function ampify_file_block( $block_content, $block ) {
+		if ( empty( $block['attrs']['displayPreview'] ) || '.pdf' !== substr( $block['attrs']['href'], -4 ) ) {
+			return $block_content;
+		}
+
+		add_action(
+			'wp_print_footer_scripts',
+			static function () {
+				if ( wp_script_is( 'wp-block-library-file' ) ) {
+					wp_dequeue_script( 'wp-block-library-file' );
+				}
+			},
+			0
+		);
+
+		// In Twenty Twenty the PDF embed fails to render due to the parent of the embed having
+		// the style rule `display: flex`. Overriding the display to `block` seems to do the trick.
+		return preg_replace(
+			'/(?<=<div\s)[^>]*class=(?:"|"[^"]*\s)wp-block-file(?:"|\s[^"]*")[^>]*>/',
+			'style="display: block;" $0',
+			$block_content
+		);
```

Comment:
This appears to not not be ideal as the flexbox was used to make the Download button appear on the right, which now doesn

Non-AMP | AMP
-----------|------
![image](https://user-images.githubusercontent.com/134745/116162018-be75b100-a6a9-11eb-884d-0d906f7bac34.png) | ![image](https://user-images.githubusercontent.com/134745/116161962-a3a33c80-a6a9-11eb-9947-81087d4ae911.png)

It seems the more fundamental issue is that the `amp-google-document-embed` doesn't have `width:100%`. I'm gonna try that approach instead.

Reply:
There, with 7cef7538e2583024fef4805309957de9cf34f8f5:

Non-AMP | AMP
-----------|------
![image](https://user-images.githubusercontent.com/134745/116162018-be75b100-a6a9-11eb-884d-0d906f7bac34.png) | ![image](https://user-images.githubusercontent.com/134745/116162647-15c85100-a6ab-11eb-852f-2a4839c17076.png)
![image](https://user-images.githubusercontent.com/134745/116162706-35f81000-a6ab-11eb-97c4-c038def27764.png) | ![image](https://user-images.githubusercontent.com/134745/116162753-47411c80-a6ab-11eb-957a-cb6a54accc09.png)